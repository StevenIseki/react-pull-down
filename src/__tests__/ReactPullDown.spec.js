/* setup enzyme */
import { configure } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
configure({ adapter: new Adapter() })

/* setup jsdom */
var jsdom = require('jsdom')
const { JSDOM } = jsdom
const window = new JSDOM('').window
global.window = window
global.document = window.document

import React from 'react'
import renderer from 'react-test-renderer'
import { shallow } from 'enzyme'
import PullDown from '../ReactPullDown'

test('PullDown renders correctly and matches snapshot', () => {
  const component = renderer.create(<PullDown />)

  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

test('PullDown renders the correct elements', () => {
  const wrapper = shallow(<PullDown />)

  expect(wrapper.find('div').length).toEqual(2)
  expect(wrapper.find('#pull-down').length).toEqual(1)
  expect(wrapper.find('svg').length).toEqual(1)
})
