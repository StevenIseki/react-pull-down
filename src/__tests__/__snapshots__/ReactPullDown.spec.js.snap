// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PullDown renders correctly and matches snapshot 1`] = `
<div
  id="pull-down"
  style={
    Object {
      "background": "#fff",
      "boxSizing": "border-box",
      "color": "#555",
      "height": "50px",
      "textAlign": "center",
      "transition": "height",
      "width": "100%",
      "zIndex": "10",
    }
  }
>
  <div
    style={
      Object {
        "height": "100%",
        "overflow": "hidden",
        "position": "relative",
      }
    }
  >
    <svg
      height="24"
      style={
        Object {
          "bottom": "10px",
          "position": "absolute",
        }
      }
      viewBox="0 0 24 24"
      width="24"
    >
      <path
        d="M0 0h24v24H0V0z"
        fill="none"
      />
      <path
        d="M7.41 8.59L12 13.17l4.59-4.58L18 10l-6 6-6-6 1.41-1.41z"
      />
    </svg>
  </div>
</div>
`;
